introspection-engine/datamodel-renderer/src/configuration/datasource.rs

```diff
@@ -137,6 +140,8 @@ impl<'a> fmt::Display for Datasource<'a> {
 
         if let Some(relation_mode) = self.relation_mode {
             writeln!(f, "relationMode = \"{}\"", relation_mode)?;
+        } else if let Some(referential_integrity) = self.referential_integrity {
+            writeln!(f, "referentialIntegrity = \"{}\"", referential_integrity)?;
```

Comment:
(just starting a quick peek, I haven't really dived into reviewing yet)

I think we wanted to deduplicate that at the datasource level — wouldn't that make the changes in this file redundant? I we should also rename `referentialIntegrity` to `relationMode` in the datasource when we reintrospect (since we do not want to maintain support for both, from the last discussions I remember on slack). So we could achieve the same and the renaming behaviour in one go by having only `relation_mode` in `Datasource` (populated from either `relationMode` or `referentialIntegrity`, with both being present as a hard error).

I am not completely up to date with the referential integrity work and it's a quick impression based on one file, I may be completely off.

Reply:
So, the "up-to-date" status is:
- we want to validate against having a co-occurrence of `relationMode` and `referentialIntegrity`: https://github.com/prisma/prisma/issues/15735
- we also want to deprecated `referentialIntegrity="prisma"|"foreignKeys"`, https://github.com/prisma/prisma/issues/15689

Dropping `referentialIntegrity` in favor of `relationMode` during re-introspection actually makes a lot of sense, I can still do that in this PR. However, `referentialIntegrity="prisma"|"foreignKeys"` must still be supported (although discouraged).

Reply:
Ok that matches my understanding. So we could delete the `Datasource::referential_integrity` field and populate `Datasource::relation_mode` with whatever is currently in `referential_integrity`. We can do the validation part (15735) later, but that would already achieve re-introspecting `referentialIntegrity` as `relationMode`, without changes in this file.

Reply:
That should be done now. I'm updating this PR's description